### PR TITLE
Add file support 

### PR DESCRIPTION
**Pull request by Jacquelinet19**
# Reliquery Pull Request
**Thank you for contributing to Reliquery.** 
Before submitting this PR please make sure:

### PR checklist
- [x] Your code builds clean without any errors or warnings
- [x] Code is formatted using the supported linter locally before pushing changes
- [x] You have added unit tests where applicable 
- [x] All tests complete and pass with changes locally
- [ ] Have updated the documentation accordingly.

### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe proposed changes
- Added support for file data type
- Added appropriate test cases

### CODE REVIEW

File: reliquery/relic.py [hunk 1]
```diff
@@ -295,6 +294,42 @@ def get_pandasdf(self, name: str) -> pd.DataFrame:
         pandas_dataframe = pd.read_json(pandas_json)
         return pandas_dataframe
 
+    def add_file_from_path(self, name: str, path: str) -> None:
+        self.assert_valid_id(name)
+
+        with open(path, "rb") as input_file:
+
+            fileSize = getsizeof(input_file)
```
Comment (AntiAnt) [inline reliquery/relic.py:302]: We should probably get the size of the file in from the available bytes in the buffer. We saw that the size of the file could be different then the actual size in bytes of the data on the file. Checkout the add_array or add_image methods on Relic, I believe there are examples there.

File: reliquery/storage.py [hunk 2]
```diff
@@ -308,7 +302,7 @@ def put_metadata(self, path: Storage, metadata: Dict):
         )
 
     def get_metadata(self, path: StoragePath, root_key: str) -> Dict:
-        dirs = ["arrays", "html", "text", "images", "json", "pandasdf"]
+        dirs = ["arrays", "html", "text", "images", "json", "pandasdf", "file"]
```
Comment (AntiAnt) [inline reliquery/storage.py:305]: Are we using "file" or "files" I see it used differently. This is a good example of why we should have this list saved in a central location. That can be saved for another pull request, but I would figure out which version to use here.

File: requirements.txt [hunk 3]
```diff
@@ -4,4 +4,5 @@ black
 uvicorn
 boto3
 flake8
-pandas
\ No newline at end of file
+pandas
+starlette
```
Comment (AntiAnt) [inline requirements.txt:8]: Why are we adding this to the requirements.txt?

File: setup.py [hunk 4]
```diff
@@ -8,6 +8,7 @@
     "flake8",
     "sphinx",
     "pandas",
+    "starlette",
```
Comment (AntiAnt) [inline setup.py:11]: Same as above: Why are we adding this to the build?

**Review by AntiAnt — CHANGES_REQUESTED** [review]

**Comment by ibebrett** [comment]
Nice comments and PR!

File: reliquery/relic.py [hunk 5]
```diff
@@ -298,9 +298,8 @@ def add_file_from_path(self, name: str, path: str) -> None:
         self.assert_valid_id(name)
 
         with open(path, "rb") as input_file:
-
-            fileSize = getsizeof(input_file)
             buffer = BytesIO(input_file.read())
+            fileSize = buffer.getbuffer().nbytes
```
Comment (ibebrett) [inline reliquery/relic.py:302]: This is an improvement, but we should make a TODO at some poitn to avoid read()ing the entire file into memory. In theory the we should be able to push binary objects that represent a stream of information. The file may be very large so reading it into memory can cause issues some times.

I'll make an issue.

**Review by ibebrett — APPROVED** [review]

File: reliquery/relic.py (hunk 5 above)
Comment (AntiAnt) [reply reliquery/relic.py:302]: ok thanks

**Review by AntiAnt — APPROVED** [review]
awesome!
